pkg/ride/generate/internal/constructors_generation.go

```diff
@@ -43,31 +44,36 @@ func extractConstructorArguments(args []actionField) ([]actionField, error) {
 	return arguments, nil
 }
 
-func checkListElementsTypes(cd *Coder, constructorName string, topListVarName string, info *listTypeInfo) {
-	var helper func(listVarName string, info *listTypeInfo)
-
-	helper = func(listVarName string, info *listTypeInfo) {
-		cd.Line("for _, elem := range %s {", listVarName)
-		cd.Line("switch t := elem.(type) {")
-
-		onelineTypes := make([]string, 0, len(info.ElementTypes()))
-		for _, tInfo := range info.ElementTypes() {
-			switch t := tInfo.(type) {
-			case *listTypeInfo:
-				cd.Line("case %s: ", t.String())
-				helper("t", t)
-			default:
-				onelineTypes = append(onelineTypes, t.String())
+func checkListElementsTypes(cd *Coder, constructorName string, topListVarName string, list stdlib.ListType) {
+	var helper func(listVarName string, ti stdlib.Type)
+
+	helper = func(listVarName string, t stdlib.Type) {
+		cd.Line("for _, el := range %s {", listVarName)
+		cd.Line("switch el.(type) {")
+
+		switch tt := t.(type) {
+		case stdlib.SimpleType:
+			cd.Line("case %s:", getType(tt))
+		case stdlib.UnionType:
+			cd.Line("case %s:", getUnionType(tt))
+		case stdlib.TupleType:
+			l := len(tt.Types)
+			cd.Line("case tuple%d:", l)
+			for i, elt := range tt.Types {
+				cd.Line("if _, ok := (el.(tuple%d)).el%d.(%s); !ok {", l, i+1, getType(elt))
```

Comment:
Why do we need type assertion if we already know what type we have? (For example, here we have `tuple%d` and we can use it).

Reply:
Here we are checking the types of elements of the tuple.